[Bug]: Milvus cannot compile after knowhere rename Timer.h

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: b90fa5f
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

Compilation fails

### Expected Behavior

Compile succeeds

### Steps To Reproduce

_No response_

### Milvus Log

```
/home/silverxia/workspace/milvus/internal/core/src/index/VectorMemIndex.cpp:26:10: fatal error: knowhere/comp/Timer.h: No such file or directory
   26 | #include "knowhere/comp/Timer.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [src/index/CMakeFiles/milvus_index.dir/build.make:104: src/index/CMakeFiles/milvus_index.dir/VectorMemIndex.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
/home/silverxia/workspace/milvus/internal/core/src/index/VectorMemNMIndex.cpp:24:10: fatal error: knowhere/comp/Timer.h: No such file or directory
   24 | #include "knowhere/comp/Timer.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

```

### Anything else?

_No response_

--- a/internal/core/src/index/VectorMemIndex.cpp
+++ b/internal/core/src/index/VectorMemIndex.cpp
@@ -23,7 +23,7 @@
 #include "config/ConfigKnowhere.h"
 
 #include "knowhere/factory.h"
-#include "knowhere/comp/Timer.h"
+#include "knowhere/comp/time_recorder.h"
 #include "common/BitsetView.h"
 #include "common/Slice.h"
 #include "common/Consts.h"

--- a/internal/core/src/index/VectorMemNMIndex.cpp
+++ b/internal/core/src/index/VectorMemNMIndex.cpp
@@ -21,7 +21,7 @@
 #include "log/Log.h"
 
 #include "knowhere/factory.h"
-#include "knowhere/comp/Timer.h"
+#include "knowhere/comp/time_recorder.h"
 #define RAW_DATA "RAW_DATA"
 
 namespace milvus::index {

